src/main/scala/shine/C/Compilation/CodeGenerator.scala

```diff
@@ -545,7 +545,7 @@ class CodeGenerator(val decls: CodeGenerator.Declarations,
         case shine.DPIA.Types.f32 => C.AST.Type.float
         case shine.DPIA.Types.f64 => C.AST.Type.double
         case _: shine.DPIA.Types.IndexType => C.AST.Type.int
-        case _: shine.DPIA.Types.VectorType => throw new Exception("Vector types in C are not supported")
+        case  _ => throw new Exception(s"$b types in C are not supported")
```

Comment:
We should avoid catch-all patterns, as they will silently match against all future extensions of the type of the value we are matching apart here.
Can we revert this change, please.